### PR TITLE
feat: stabilize kernel interface

### PR DESCRIPTION
**Pull request by stlankes**
In case, that POSIX provides similar system calls, HermitOS uses variants of these system calls. These similiarties should increase the readability of the interface.

See also hermit-os/hermit-rs#570

### CODE REVIEW

File: src/syscalls/socket.rs [hunk 1]
```diff
@@ -101,15 +100,15 @@ pub struct sockaddr_in {
...
-			let address = IpAddress::v4(
-				addr.sin_addr.s_addr[0],
-				addr.sin_addr.s_addr[1],
-				addr.sin_addr.s_addr[2],
-				addr.sin_addr.s_addr[3],
-			);
+			#[cfg(target_endian = "big")]
+			let s_addr = addr.sin_addr.s_addr.to_ne_bytes();
+			#[cfg(target_endian = "little")]
+			let s_addr = addr.sin_addr.s_addr.swap_bytes().to_ne_bytes();
```
Comment (mkroening) [inline src/syscalls/socket.rs:109]:
```suggestion
			let s_addr = addr.sin_addr.s_addr.to_be_bytes();
```

File: src/syscalls/socket.rs [hunk 2]
```diff
@@ -120,12 +119,8 @@ impl From<sockaddr_in> for IpListenEndpoint {
 impl From<sockaddr_in> for IpEndpoint {
 	fn from(addr: sockaddr_in) -> IpEndpoint {
 		let port = u16::from_be(addr.sin_port);
-		let address = IpAddress::v4(
-			addr.sin_addr.s_addr[0],
-			addr.sin_addr.s_addr[1],
-			addr.sin_addr.s_addr[2],
-			addr.sin_addr.s_addr[3],
-		);
+		let s_addr = addr.sin_addr.s_addr.to_ne_bytes();
```
Comment (mkroening) [inline src/syscalls/socket.rs:122]:
```suggestion
		let s_addr = addr.sin_addr.s_addr.to_be_bytes();
```

File: src/syscalls/socket.rs [hunk 3]
```diff
@@ -136,14 +131,15 @@ impl From<IpEndpoint> for sockaddr_in {
 	fn from(endpoint: IpEndpoint) -> Self {
 		match endpoint.addr {
 			IpAddress::Ipv4(ip) => {
-				let mut in_addr: in_addr = Default::default();
-				in_addr.s_addr.copy_from_slice(ip.as_bytes());
+				let sin_addr = in_addr {
+					s_addr: u32::from_ne_bytes(ip.as_bytes().try_into().unwrap()),
```
Comment (mkroening) [inline src/syscalls/socket.rs:135]:
```suggestion
					s_addr: u32::from_be_bytes(ip.as_bytes().try_into().unwrap()),
```

File: src/time.rs [hunk 4]
```diff
@@ -1,21 +1,26 @@
 use crate::arch;
 
+#[allow(non_camel_case_types)]
+pub type time_t = i64;
+#[allow(non_camel_case_types)]
+pub type suseconds_t = i32;
```
Comment (mkroening) [inline src/time.rs:6]:
```suggestion
pub type suseconds_t = u32;
```

File: src/time.rs [hunk 5]
```diff
@@ -36,20 +41,20 @@ pub struct itimerval {
...
-#[derive(Copy, Clone, Debug)]
+#[derive(Copy, Clone, Debug, Default)]
 #[repr(C)]
 pub struct timespec {
 	/// seconds
-	pub tv_sec: i64,
+	pub tv_sec: time_t,
 	/// nanoseconds
-	pub tv_nsec: i64,
+	pub tv_nsec: i32,
```
Comment (mkroening) [inline src/time.rs:50]:
```suggestion
	pub tv_nsec: u32,
```

**Review by mkroening — CHANGES_REQUESTED** [review]
Looks good to me, aside from a few remarks. 👍

File: src/time.rs (hunk 5 above)
Comment (stlankes) [reply src/time.rs]: I revised it already

**Review by mkroening — APPROVED** [review]

**Pull request merged**